modules/preview-react/radio/lib/RadioInput.tsx

```diff
@@ -0,0 +1,206 @@
+    },
+
+    '&:focus:after, &:focus:hover:after': {
+      outline: 'transparent',
+
+      ...focusRing({
+        width: variant === 'inverse' ? 2 : 1,
+        separation: 0,
+        animate: false,
+        innerColor: variant === 'inverse' ? colors.blackPepper400 : colors.frenchVanilla100,
+        outerColor: variant === 'inverse' ? colors.frenchVanilla100 : colors.blueberry400,
+      }),
+    },
+
+    '&:focus:checked:after, &:focus:hover:checked:after': {
+      outline: 'transparent',
+      ...focusRing({
+        width: 2,
+        separation: 2,
+        animate: false,
+        innerColor: variant === 'inverse' ? colors.blackPepper400 : colors.frenchVanilla100,
+        outerColor: variant === 'inverse' ? colors.frenchVanilla100 : colors.blueberry400,
+      }),
+    },
+
+    ...mouseFocusBehavior({
+      '&:focus:after, &:focus:hover:after, &:focus:active:after, &:focus:hover:checked:after, &:focus:active:checked:after': {
+        ...focusRing({
+          width: 0,
+          outerColor: variant === 'inverse' ? colors.frenchVanilla100 : themeFocusOutline,
+        }),
+      },
+    }),
+  })
+);
+
+const RadioInputWrapper = styled(Flex)<Pick<RadioLabelProps, 'disabled' | 'variant'>>(
+  {
+    // Hover Ripple element
+    '::before': {
+      content: "''",
+      borderRadius: borderRadius.circle,
+      height: radioHeight,
+      transition: 'box-shadow 150ms ease-out',
+      width: radioWidth,
+      pointerEvents: 'none',
+    },
+  },
+  ({variant, disabled}) => ({
+    '::before': {
+      opacity: variant === 'inverse' ? '0.4' : '1',
+    },
+    '&:hover:before': {
+      boxShadow: disabled ? undefined : `0 0 0 ${rippleRadius}px ${colors.soap200}`,
+    },
+  })
+);
+
+const useRadioInput = createElemPropsHook(useRadioModel)(
+  (model, ref, elemProps: {value?: string; checked?: boolean} = {}) => {
+    const {disabled, variant} = React.useContext(RadioLabelContext);
+    return {
+      'aria-describedby': model.state['aria-describedby'],
+      disabled: disabled,
+      variant: variant,
+      checked: elemProps.value === model.state.value,
+      'aria-checked': elemProps.value === model.state.value,
+      onChange(event: React.ChangeEvent<HTMLInputElement>) {
+        model.onChange(event);
+      },
+      name: model.state.name,
+    };
+  }
+);
+export const RadioInput = createSubcomponent('input')({
+  modelHook: useRadioModel,
+  elemPropsHook: useRadioInput,
+})<RadioLabelProps>(
+  (
+    {
+      children,
+      disabled,
+      checked,
+      value,
+      name,
+      variant,
+      'aria-checked': ariaChecked,
+      'aria-describedby': ariaDescribedby,
+      ...elemProps
+    },
+    Element,
+    model
+  ) => {
+    return (
+      <RadioInputWrapper
+        height="18px"
+        width="18px"
+        flex="0 0 auto"
+        disabled={disabled}
+        variant={variant}
+        {...elemProps}
+      >
+        <StyledRadioInput
```

Comment:
I noticed RadioInput.tsx uses its own copies of `StyledRadioInput` and `RadioInputWrapper` separate from `StyledRadioInput` and `RadioInputWrapper` in StyledRadio.tsx. At a glance, the two components look identical in both files. Can we have RadioInput.tsx use the `StyledRadioInput` and `RadioInputWrapper` from StyledRadio.tsx instead of duplicating that code in both places?

Or, better yet, can we have RadioInput.tsx use `StyledRadio` directly, much like how [MenuItem uses `StyledMenuItem` directly](https://github.com/Workday/canvas-kit/blob/master/modules/react/menu/lib/MenuItem.tsx#L182)?

Reply:
Ended up using the exported styled elements!